packages/host/app/components/operator-mode/card-schema-editor.gts

```diff
@@ -184,6 +233,14 @@ export default class CardSchemaEditor extends Component<Signature> {
     ).includes(fieldName);
   }
 
+  @action
+  calculateTotalOwnFields() {
+    this.totalOwnFields = this.args.cardType.fields.filter((field) =>
+      this.isOwnField(field.name),
+    ).length;
+    this.args.setTotalFieldsOfSchema?.(this.totalOwnFields);
```

Comment:
This feels awkward, sending data up from many different components to calculate the number of total fields. We need a single source of truth for calculating this - take a look at `loadInheritanceChain` where the whole inheritance chain data structure gets created - there you can traverse the items. Each item has a `cardType` property that comes with `fields`. I think you can use that to reduce it to the total field count.

Reply:
I've considered doing that, but it seems we're going to have a redundant function such as `isOwnField` on `CardSchemaEditor` and `CardAdoption`. Or can we calculate the `totalFields` based on the top items of `cardInheritanceChain`?

Reply:
Yeah it looks like the top parent in the card type has all fields from the chain, maybe you can simply take that

Reply:
After I checked it, the last child doesn't inherit all fields from the chain. So I decided to make `schema-editor` utils and put `isOwnField` and `calculateTotalOwnFields` functions there. So we can use those functions in `CardSchemaEditor` and `CardAdoption` components. [63e5412](https://github.com/cardstack/boxel/pull/701/commits/63e5412ac10caa2cb7034b49c9fdd2cf78fc169c)